src/shared-components/optionButton/style.ts

```diff
@@ -2,28 +2,43 @@ import styled from '@emotion/styled';
 import { css } from '@emotion/core';
 
 import { style as TYPOGRAPHY_STYLE } from '../typography';
-import { COLORS, ANIMATION, SPACER, BOX_SHADOWS } from '../../constants';
-import Container from '../container';
+import {
+  COLORS, ANIMATION, SPACER, BOX_SHADOWS, 
+} from '../../constants';
+import { containerStyles, ContainerType } from '../container/style';
 
-const getOptionTypeStyles = optionType => {
+import { OptionButtonProps } from '.';
+
+type BaseIconWrapperStylesProps = Required<
+  Pick<OptionButtonProps, 'buttonType' | 'optionType' | 'selected'>
+>;
+
+const getOptionTypeStyles = (
+  optionType: BaseIconWrapperStylesProps['optionType'],
+) => {
   if (optionType === 'checkbox') {
     return css`
       border-radius: 4px;
     `;
   }
+
   return css`
     border-radius: 50%;
   `;
 };
 
-const getTypeColor = type => {
-  if (type === 'secondary') {
+const getTypeColor = (buttonType: BaseIconWrapperStylesProps['buttonType']) => {
+  if (buttonType === 'secondary') {
     return COLORS.secondary;
   }
+
   return COLORS.primary;
 };
 
-export const ClickableContainer = styled(Container)`
+export const ClickableContainer = styled.button<{
```

Comment:
I'm pretty sure my preference is for us to use `styled.button` directly, for explicitness. However, There is a version of this PR that could rely on [`withComponent`](https://emotion.sh/docs/styled#change-the-rendered-tag-using-withcomponent) instead. The simplicity could be arguably more clear. 

If we prefer to rely on more emotion API to DRY things up on our end, I can explore that in more detail next week.

Reply:
I think I like using the `button` as well. My big takeaway from the past two weeks is that elements should be defined by what they do, not what they look like. Using the `button` here and then applying the styles keeps these concerns well separated.